api/v1/services/AuthService.js

```diff
@@ -65,3 +68,48 @@ module.exports.verify = function(token) {
 			throw new errors.UnprocessableRequestError(message);
 		});
 };
+
+
+/**
+ * Generates a token and deletes all existing tokens
+ * with the same scope.
+ * @param {User} user The user object to create a reset token for.
+ * @param {String} scope The scope to create the token for.
+ * @return {Promise<Bool>} Returns a Promise that resolves to
+ *                         true on a successful token creation.
+ */
+module.exports.generateToken = function(user, scope){
+	var tokenVal = utils.crypto.generateResetToken();
+	var user_id = user.get('id');
+
+	return Token
+		.where({user_id: user_id, type: scope}).fetchAll()
+		.then(function(tokens) {
+			return tokens.invokeThen('destroy')
+				.then(function() {
+					var token = Token.forge({type: scope, value: tokenVal, 'user_id': user_id});
+					logger.info('Password recovery token generated:' + tokenVal);
+					return token
+						.validate()
+						.catch(Checkit.Error, utils.errors.handleValidationError)
+						.then(function(validated) {
+							return token.save();
+						})
+						.then(function () { 
+							return tokenVal; 
+						});
+				});
+		});
+};
+
+/**
+ * Resets the User's password given that the token is the same as the one
+ * generated when the user requests a password change
+ * @param {Token} token the Token user has obtained to reset password
+ * @param {String} password the password the User would like to change it to
+ * @return {Promise} resolving to the validity of the provided password
+ * @throws {InvalidParameterError} when the requested token cannot be found
+ */
+module.exports.resetPassword = function(token, password) {
+	return Promise.resolve(token.resetUserPassword(password));
```

Comment:
Is this `resetUserPassword` defined anywhere? I feel like this entire method is a `UserService` method that should take a token and a new password. That would mean it'd just basically be a wrapped "set password" call with a token that is deleted by a `TokenService`